---
layout: post
title: "Genome-wide Association Study (GWAS) in TASSEL (GUI)"
tags: [GWAS,TASSEL, GUI, Genome-wide association test, GLM, MLM, Linear model]
image: /image/gwas/gwaslogo.png
share-img: /image/gwas/gwaslogo.png
---


__TASSEL__ aslo known as `Trait Analysis by aSSociation, Evolution and Linkage` is a powerful statistical software to conduct `association mapping` such as `General Linear Model (GLM)` and `Mixed Linear Model (MLM)`. The GUI (graphical user interface) version of TASSEL is very well built for anyone who does not have a background or experience in working in `command line`. In this tutorial, I will show how to prepare `input` files and run assoication analysis in TASSEL. For detailed information on TASSEL, user's guide and further documentation please visit: 
`https://www.maizegenetics.net/tassel`

{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** Please download a `sample phenotype` and `genotype data` (.hmp.txt format) from the below link:

- <a href="/image/gwas/sample_Genotype-Chr-1_data_forTASSEL.hmp.txt"> Download Sample Genotype data </a>
- <a href="/image/gwas/sample_Phenotype_data_forTASSEL.txt"> Download Multiple Phenotype data in TASSEL format </a>


<h1>1.1 Download and install TASSEL software </h1>
Download and install the latest version of the <strong> TASSEL software </strong> at this link:
`https://www.maizegenetics.net/tassel`

<hr>

<h2>1.2: Preparing the Input files </h2>
<h2> Phenotype  file</h2>
Prepare the phenotype file as shown below in the figure

{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** Please remember if your data has `covariates` such as `sex`, `age` or `treatment`, then, please categories them with header name `factor`.

<center><img src="/image/gwas/pheno.png" alt="Phenotype data"></center>

<hr>

<h2> Genotype file</h2>
TASSEL allows various genotype file formats such as `VCF` (variant call format), `.hmp.txt`, and `plink`. In this tutorial, I am using the `hmp.txt` version of the genotype file. The below githe screenshot of the hmp.txt genotype file. 

<center><img src="/image/gwas/geno.JPG" alt="Genotype data"></center>
<hr>

<h1> Step 1.2: Importing phenotype and genotype files </h1>
Import the files by following the steps shown below.
__Tip!__ Both files can be opened at same time holding `CTRL` and clicking the file names. 

<center><img src="/image/gwas/importfiles.gif" alt="Import data"></center>

<hr>

<h2> 1.3 Phenotype distribution plot</h2>
It is always a wise idea to look at the phenotype distribution by plotting to check for any outliers. Follow below steps to plot histogram of your phenotype data.

<center><img src="/image/gwas/phenoplot.gif" alt="Phenotype distribution"></center>
<hr>

<h2> 1.4 Genotype summary analysis </h2>
Next crucial step is to look at the genotype data by simply following the steps shown.
Couple of keys things to look at are:
<ol>
  <li>Minor allele frequency distribution </li>
  <li>Missing genotypic data to see if it requires to be imputed </li>
  <li>Proportion of heterozygous in the samples to check for self-ed samples </li>
</ol>

<center><img src="/image/gwas/genosummary.gif" alt="Genotype summary"></center>
<hr>

  
<h1> 2.0 Conduct GWAS analysis </h1>

<h3> 2.1 multidimensional scaling (MDS) </h3>
MDS output can be used as the covariate in the GLM or MLM to correct for population structure.
Please follow the steps shown below:

<center><img src="/image/gwas/mds.gif" alt="MDS"></center>
  
<hr>

<h3> 2.2 Intersecting the files </h3>
Intersect the `genotype`, `phenotype` and `MDS` files by following the steps below:

<center><img src="/image/gwas/intersect.gif" alt="Intersect files"></center>

<hr>

<h1> 3.0 running General Linear Model (GLM) </h1>
Run the GLM analysis by selecting the `intersected` files following the steps below:

<center><img src="/image/gwas/glm.gif" alt="GLM"></center>
<hr>

The output of the GLM analyis is produced ubder the `Result` node. The GLM association test can be evaluated by plotting `Q-Q plot` and the `Manhattan` plot as shown below. 

<center>
<img src="/image/gwas/manhattan.png" width="auto" height="200" alt="GLM Manhattan plot">
<img src="/image/gwas/qqplot.png" width="auto" height="200" alt="Q-Q plot">
</center>

From the above Q-Q plot, we can see that are several markers that appear to be falsely associated with the trait, therefore, to control this confounding effect, use `Kinship` matrix as an another covariate in the linear model

<hr>

<h1> 4.0 Mixed Linear Model (MLM)  </h1>


<h3> 4.1 Calculating Kinship matrix </h3>
Follow the below steps to calcuate the kinship matrix:

<center><img src="/image/gwas/kinship.gif"  alt="Kinship matrix"></center>
<hr>

<h3> 4.2 running Mixed Linear Model (MLM)  </h3>
MLM model includes the PCA and the kinship matrix i.e. MLM(PCA+K).

Therefore, once the Kinship matrix has been calculated, MLM can be now be conducted by following below steps:

<center><img src="/image/gwas/mlm.gif"  alt="MLM steps"></center>
<br>

Plot the output (MLM stats file in the Results branch following the above shown steps).
<center>
<img src="/image/gwas/manhattanMlm.png" width="auto" height="200" alt="MLM Manhattan plot">
<img src="/image/gwas/qqmlm.png" width="auto" height="200" alt="MLM Q-Q plot">                                                           
</center>

<hr>

<h3> 4.2 Exporting results  </h3>
One may export the results in .txt format by the following the below steps:

<center><img src="/image/gwas/export.gif"  alt="Export results">
</center>

<hr>

<h3> 4.3 Significance Threshold  </h3>
`Bonferroni threshold` can be determined to identify significantly markers associated with the trait  by using the below equation:
```console
P ≤ 1/N (α =0.05)

For example:
If there are 13,169 sites (markers), then Bonferroni threshold would be = -log10(0.05/13169) = 5.42058279333

```
where, N is the total number of markers tested in association analysis) was used to identify the most significantly markers associated with the trait. Similarly, another way is to perform `FDR` (`False Discoveyy Rate`) correction method.

<h3> 4.4 Plotting results in R </h3>

The GLM and/or MLM stats can be plotted in R using the `qqman` package. Read more about the package <a href="https://cran.r-project.org/web/packages/qqman/vignettes/qqman.html">here</a>

R code to plot: 

```powershell

library(qqman)

# import data
mlm_stats <- read.table("GLMstats.txt", header = T, sep = "\t")


# Manhattan plot
manhattan(mlm_stats, chr="Chr", bp="Pos", snp="Marker", p="p", annotateTop = T,
          genomewideline = 5.42, suggestiveline = F)

# QQ plot
qq(mlm_stats$p)

```

Output:
<center>
  <img src="/image/gwas/manhattanPlot_qqman.png"  alt="QQman results">
</center>

<h1> 5.0 Estimation and plotting of LD decay over distance from the LD results from TASSEL </h1>
Steps are shown below:
<center>
  <img src="/image/gwas/LDdecay.gif"  alt="LD decay plot">
</center>

Code to plot in R using GGplot: 

```powershell

library(ggplot2)
library(ggridges)
theme_set(theme_ridges())

ld_chr1 <- read.table("LD_chr_1_50window.txt", header = T, na.strings = "NaN")

ggplot() +
  geom_point(aes(ld_chr1$Dist_bp/1000000, ld_chr1$R.2), na.rm = T, color ="blue", alpha=0.05) +
  labs(x="Chromosome 1 - Physical distance (Megabases)", y=expression(LD~(r^{2})))
  
```

Output of the LD decay plot for chromosome 1
<center>
  <img src="/image/gwas/LD_decay_plot_chr_1.png"  alt="LD decay plot">
</center>

<center><h> --- End of Tutorial --- </h3></center>


__Thank you__ for reading this tutorial. If you have any questions or comments, please let me know in the comment section below or send me an email. 


<h3> Bibliography </h3>
<p>
Bradbury PJ, Zhang Z, Kroon DE, Casstevens TM, Ramdoss Y, Buckler ES. (2007) TASSEL: Software for association mapping of complex traits in diverse samples. Bioinformatics 23:2633-2635.
</p>
