---
layout: post
title: "Pruning genetic markers based on their physical distance and linkage disequilibrium (LD)"
tags: [Genetic Markers, Pruning, PLINK, TASSEL, GUI, Linkage disequilibrium]
image: /image/prune/prune.png
share-img: /image/prune/prune.png
---


High density markers do not provide any additional information, therefore, can be pruned based on the physical distances between adjacent markers and linkage disequilibrium (LD). In this tutorial, I will show how to prune markers based on their physical position in `TASSEL` software, and based on LD in `PLINK` software. 

<h1>Pruning markers based on the physical distances in TASSEL</h1>

For detailed information on how to use `TASSEL` software, please consult user's guide and further documentation at: 
`https://www.maizegenetics.net/tassel`

<h2>1.1 Download and install TASSEL software </h2>
Download and install the latest version of the <strong> TASSEL software </strong> at this link:
`https://www.maizegenetics.net/tassel`

<hr>

<h2> Genotype file</h2>
TASSEL allows various genotype file formats such as `VCF` (variant call format), `.hmp.txt`, and `plink`. In this tutorial, I am using the `hmp.txt` version of the genotype file. The below is the screenshot of the hmp.txt genotype file. 

<center><img src="/image/gwas/geno.JPG" alt="Genotype data"></center>
<hr>

<h2> Step 1.2: Importing files </h2>
Import the files by following the steps shown below.
__Tip!__ Both files can be opened at same time holding `CTRL` and clicking the file names. 

<center><img src="/image/gwas/importfiles.gif" alt="Import data"></center>

<hr>

<h2> Command line pruning in TASSEL</h2>
Use below sample command line to filter :

```powershell
./run_pipeline.pl -importGuess /Users/lcj34/genotyep.hmp.txt -ThinSitesByPositionPlugin 
 -o /Users/lcj34/thin40000.vcf -minDist 40000 –endPlugin
```
<hr>


<h2>Pruning in GUI of TASSEL </h2>
Or else, in TASSEL GUI, you can use “Thin Sites by Position” plugin:


<center><img src="/image/prune/prunetassel.gif" alt="prune in TASSEL"></center>
<hr>

<h1> 2.0 LD based pruning in PLINK software </h1>

<h2>1.1 Download and install PLINK software </h2>
Download and install the latest version of the <strong> PLINK software </strong> at this link:
`http://zzz.bwh.harvard.edu/plink/download.shtml`

{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** While downloading the software make sure you choose your correct platform!

<h2> 2.1 Converting your genotype data into PLINK format </h2>
If you have a genotype data in formats such as VCF, then you will need to convert it to PLINK format in `VCF tools` using the command line below:

```powershell
vcftools --vcf myvcf.vcf --plink --out myplink
```
{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** Your genotype file has to be in the VCF format in order to convert into PLINK format

<center><img src="/image/prune/vcftoplink.gif" alt="convert vcf to plink"></center>

<hr>

<h2> 2.2 Extracting markers for pruning based on LD in PLINK </h2>
PLINK has two options for LD thinning/pruning: based on `variance inflation factor` (by regressing a SNP on all other SNPs in the window simultaneously) and based on `pairwise correlation (R2)`.  These are the `--indep` and `--indep-pairwise` options, respectively. 

Below is the code:

```powershell
plink --noweb --file data_in --indep 50 5 2 --out data_out
```
The command above that specifies 50 5 0.5 would 
<ul>
<li> consider a window of 50 SNPs </li>
<li> calculate LD between each pair of SNPs in the window</li>
<li> remove one of a pair of SNPs if the LD is greater than 0.5</li>
<li> shift the window 5 SNPs forward and repeat the procedure</li>
</ul>
<hr>
<center><img src="/image/prune/pruning.gif" alt="pruning by LD"></center>
<hr>

The output of the above commands creates two lists of SNPs: those that are pruned out and those that are not. See below:

```powershell
 plink.prune.in
 plink.prune.out
```
<hr>
<h2> 2.3 Make a new, pruned file </h2>

Next, make a new, pruned file using the command below:
```powershell
plink --file data --extract plink.prune.in --make-bed --out pruneddata
```
<center><img src="/image/prune/newfileprune.gif" alt="new pruned file"></center>


<h2> 2.4 Convert the prunned output file into VCF </h2>

The pruned file can be converted back to VCF format using the command below:
```powershell
plink109 --bfile pruneddata --recode vcf --out vcf_pruned
```
{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** please remember this parameter is only available in PLINK v 1.09

<img src="/image/prune/prunedfinal.gif" alt="final pruned file in VCF">

Once the final pruned data is converted to `VCF` file it can be viewed in TASSEL software for further analysis.

<hr>

<center><h3> --- End of Tutorial --- </h3></center>


__Thank you__ for reading this tutorial. If you have any questions or comments, please let me know in the comment section below or send me an email. 


<h3> Bibliography </h3>
<p>
Package:     PLINK (including version number)
        Author:      Shaun Purcell
        URL:         http://pngu.mgh.harvard.edu/purcell/plink/

        Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, 
	Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ & Sham PC (2007) 
        PLINK: a toolset for whole-genome association and population-based 
	linkage analysis. American Journal of Human Genetics, 81.
</p>
