---
layout: post
title: "How to use Multidimensional Scaling (MDS) to quality control your genetic data?"
tags: [Multidimensional Scaling, MDS, Genetic data, QC]
image: /image/mds/plotmds.png
share-img: /image/mds/plotmds.png
---

__Multidimensional Scaling (MDS)__ statistical method can be effectively used to elucidate hidden population structure, and more importantly, as a quality control analysis tool while working with genetic data. I strongly encourage anyone to QC their data (phenotype or genotype data) before proceeding to any further analysis. In my own research, I work with plethora of genetic data genetrated from the grape breeding programs, and I use MDS religiously to identify any possible pollen contamination (selfed or out-crossess), and or mislabled samples, and further to use it as a population structure covariate in GLM and MLM models in GWAS. In this tutorial, one can use the you through how you can use this method in your own data analysis in `TASSEL` software and visualize in `R` software. For detail information on MDS, please read this article at this <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-179">link</a>

<h2> What is MDS? </h2>
MDS is also known as `Prinicpal Co-ordinate Analysis (PCoA)`, and produces results that are very similar to `Principal Component Analysis (PCA)`. A genome-wide pairwise `Identity-By-State (IBS)` distance matrix using the genotype data is first calculated, then following the MDS analysis. 

<img src="/image/mds/mdsanimation1.gif" alt="3D MDS plot" width="128" height="128">

<h2> Calculating MDS </h2>
In this tutorial, I will show how to calculate MDS in `TASSEL` software. If you are familiar with TASSEL or do not have it installed on your computer, then, please download and go through its documentation at this <a href="https://www.maizegenetics.net/tassel"> link </a>. 

<h3> Step: 1 </h3>
Import data in TASSEL.
<img src="/image/mds/tassel1.gif" alt="Import data" width="128" height="128">

<h3> Step: 2 </h3>
Calculate distance matrix.
<img src="/image/mds/tassel2.gif" alt="Calculate distance matrix" width="128" height="128">

<h3> Step: 3 </h3>
Calculate MDS.
<img src="/image/mds/tassel3.gif" alt="Calculate MDS" width="128" height="128">

<h3> Step: 4 </h3>
Plot PCoAs in TASSEL.
<img src="/image/mds/tassel4.gif" alt="Plot MDS" width="128" height="128">



<center><h3> --- End of Tutorial --- </h3></center>

__Thank you__ for reading this tutorial. I really hope this helpful in giving you the concept and technology behind <strong>AmpSeq </strong> and the data analysis. If you have any questions or comments, please let comment below or send me an email. 
<form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
<input type="hidden" name="cmd" value="_donations" />
<input type="hidden" name="business" value="8ZF7YRTZ42EKU" />
<input type="hidden" name="item_name" value="To support education for all." />
<input type="hidden" name="currency_code" value="USD" />
<input type="image" src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif" border="0" name="submit" title="PayPal - The safer, easier way to pay online!" alt="Donate with PayPal button" />
<img alt="" border="0" src="https://www.paypal.com/en_US/i/scr/pixel.gif" width="1" height="1" />
</form>
Happy AmpSeq-ing !


<h3> Bibliography </h3>
Bradbury, Peter J., et al. "TASSEL: software for association mapping of complex traits in diverse samples." Bioinformatics 23.19 (2007): 2633-2635.



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-123359651-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-123359651-1');
</script>

<script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script>
  (adsbygoogle = window.adsbygoogle || []).push({
    google_ad_client: "ca-pub-5126027065024936",
    enable_page_level_ads: true
  });
</script>
