---
layout: post
title: "Import SSR marker data in TASSEL software"
tags: [SSR, TASSEL, GLM, PCA]
image: /image/ssr/ssr.png
share-img: /image/ssr/ssr.png
---



In this tutorial, I would like to share how one can import `SSR` (simple sequence repeats) marker data on to `TASSEL` software, and perfrom analyses such as `imputation`, `General Linear Model` and `Principle Component Analysis`.

<h2>1.0 Installing TASSEL software</h2>

Download and install the latest version of the <strong> TASSEL software </strong> at this link:
`https://www.maizegenetics.net/tassel`

__Ignore__ this step if you already have installed TASSEL.

<hr>

<h2>1.1 Input data</h2>
The input SSR marker data must formated as instructed below:
<ul> 
<li>Add "Numeric” in the first line of the data</li>
<li>The missing data (-1) was denoted as “NA” </li>
<li>TASSEL only allows numeric marker data ranging from 0 to 1</li>
<li>Save your numeric data in a text file format (.txt) prior to loading in tassel </li>
</ul>
<br>

{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Note:** If you have two alleles per SSR marker then you can load it on TASSEL as a numeric data as `0` for major and `1` for minor alleles. However, If SSRs alleles have more than 2 alleles per marker I suggest converting as below:

<ul>
<li> Most common = A </li>
<li>2nd most common = C </li>
<li>3rd most common = G </li>
<li>4th most common = T </li>
<li>5th most common = + </li>
<li>All other alleles = - (dash) </li>
</ul>

<br>

See an example of the input file:
<img src="/image/ssr/input.png" alt="Input data">

<hr>

<h2>1.2 Importing phenotype and genotype files </h2>
Import the files by following the steps shown below.

{: .box-note}
<i class="fa fa-commenting" aria-hidden="true"></i> **Tip!** Both files can be opened at same time holding `CTRL` and clicking the file names. 

<center><img src="/image/ssr/load.gif" alt="load">
</center>
<hr>


<h2>1.3 Imputation</h2>
This is an optional step, however, in order to conduct `PCA` analysis you need to impute and should not have any missing data.

Follow the below steps to perform numerical imputation:
<center><img src="/image/ssr/impute.gif" alt="impute">
</center>

<hr>

<h2>1.4 PCA analysis</h2>

Follow the below steps to perform PCA analysis:
<center><img src="/image/ssr/pca.gif" alt="pca"></center>
<hr>

<h2>1.5 perform GLM</h2>
Run the GLM analysis by selecting the `intersected` files following the steps below:

<center><img src="/image/ssr/glm.gif" alt="GLM"></center>
<hr>

<h2>2.0 Convert Genotype calls into Numeric format in TASSEL </h2>

To convert your `genotype data` (A,C,G,T) into `numeric format` (0, 0.5, 1) follow the below steps: 

<center> <img src="/image/ssr/GenotypeToNumeric.gif" alt="Numeric Format"> </center>

<br>

<center><h3> --- End of Tutorial --- </h3></center>


__Thank you__ for reading this tutorial. If you have any questions or comments, please let me know in the comment section below or send me an email. 


<h3> Bibliography </h3>
<p>
Bradbury PJ, Zhang Z, Kroon DE, Casstevens TM, Ramdoss Y, Buckler ES. (2007) TASSEL: Software for association mapping of complex traits in diverse samples. Bioinformatics 23:2633-2635.
</p>
