---
layout: post
title: "Check for mendelian errors using pedigree and genotype information "
tags: [Genetic data, Mendelian Error, bcftools]
image: /image/bcftools/plot.png
share-img: /image/bcftools/plot.png
---

Cluster based methods such as `multidimensional scaling (MDS)` and `priniciple component analysis (PCA)` are traditionally used in identifying samples with genotypic inconsistencies, however, it is important to identify genotypes with high mendelian inconsistencies based on their pedigree and genotype information. In this tutorial, I would like to share how one can quickly check samples with high mendelian inconsistencies using genotypic data and pedigree information in `bcftools`.  

<h1>Installing bcftools</h1>

`bcftools` is only available for `Linux` operating system. For detailed information on how to install it please consult user's guide and further documentation at: 
`https://samtools.github.io/bcftools/howtos/install.html`

<hr>
<center>
<a href="https://click.linksynergy.com/fs-bin/click?id=876kEArXFCo&offerid=320088.10000211&type=4&subid=0" rel="nofollow"><IMG alt="Bake Me  A Wish" border="0" src="https://www.bakemeawish.com/banner/Newsletter/febaff/rainbow_728x90.jpg"></a><IMG border="0" width="1" height="1" src="https://ad.linksynergy.com/fs-bin/show?id=876kEArXFCo&bids=320088.10000211&type=4&subid=0">
</center>

<hr>

<h2> Input data</h2>
Two files are required as:
<ul> 
<li>Genotype file in VCF format </li>
<li>Pedigree file in text (.txt) version</li>
</ul>

<hr>
<a href="https://click.linksynergy.com/fs-bin/click?id=iVSc/qZ7CV4&offerid=595115.98&subid=0&type=4" rel="nofollow"><IMG border="0"   alt="The Trailer Parts Outlet 5% OFF coupon code SUMMER5" src="https://ad.linksynergy.com/fs-bin/show?id=iVSc/qZ7CV4&bids=595115.98&subid=0&type=4&gridnum=16"></a></center>
<hr>

<h2> Pedigree file format </h2>
The pedigree file should be formatted as shown below:
```
mother,father,child1
mother,father,child2
mother,father,child3
mother,father,child4
mother,father,child5
mother,father,child6
mother,father,child7
mother,father,child8
mother,father,child9
mother,father,child10
mother,father,child11
mother,father,child12
mother,father,child13
mother,father,child14
mother,father,child15
mother,father,child16
```

__Important!__ Make sure the parent names are sperated by comma (,) without any spaces in the text file.
<hr>
<a href="https://click.linksynergy.com/fs-bin/click?id=876kEArXFCo&offerid=635768.1456&type=3&subid=0" rel="nofollow">Find all your eBooks and audiobooks together in one place with the free Walmart eBooks App for your iOS or Android smartphone or tablet.</a><img border="0" width="1" alt="" height="1" src="https://ad.linksynergy.com/fs-bin/show?id=876kEArXFCo&bids=635768.1456&type=3&subid=0" >
<hr>

<h2> running the bcftools</h2>
Use below sample command line to run the program:

```
$ bcftools +mendelian file.vcf -T pedigree.txt -c 2> output  
```
<center>
<a href="https://click.linksynergy.com/fs-bin/click?id=iVSc/qZ7CV4&offerid=663779.16&subid=0&type=4" rel="nofollow"><IMG border="0"   alt="Blooms Today 25% Sitewide Savings" src="https://ad.linksynergy.com/fs-bin/show?id=iVSc/qZ7CV4&bids=663779.16&subid=0&type=4&gridnum=16"></a></center>
<hr>

<h2>Output</h2>
The out file contains the following headers:
```
# [1]nOK  [2]nBad  [3]nSkipped  [4]Trio
334518    4621     996          mother,father,child
```
Where,
<ul>
<li>nOK .. number of genotypes with no missing sites and no Mendelian error</li>
<li>nBad .. number of genotypes with a Mendelian error</li>
<li>nSkipped .. number of genotypes at which at least one individual missing and therefore could not be considered</li>4
</ul>


<a href="https://click.linksynergy.com/fs-bin/click?id=876kEArXFCo&offerid=683866.27&type=4&subid=0" rel="nofollow"><IMG alt="13to24" border="0" src="https://i.imgur.com/ztq9S6O.jpg"></a><IMG border="0" width="1" height="1" src="https://ad.linksynergy.com/fs-bin/show?id=876kEArXFCo&bids=683866.27&type=4&subid=0"></center>
<hr>

<h2> Plotting output </h2>
The output file can be opened directly in MS Excel and can be interpreted by plotting it. 

<center><img src="/image/bcftools/plot.png" alt="PLot output">
<a href="https://click.linksynergy.com/fs-bin/click?id=876kEArXFCo&offerid=675514.47&subid=0&type=4" rel="nofollow"><IMG border="0"   alt="TW Steel UK" src="https://ad.linksynergy.com/fs-bin/show?id=876kEArXFCo&bids=675514.47&subid=0&type=4&gridnum=16"></a>
</center>

In the above plot, it can be noted that there are four samples (in red) that have the highest mendelian error in comparison to the rest of samples, which should be excluded from any further analysis.

<hr>

<center><h3> --- End of Tutorial --- </h3></center>


__Thank you__ for reading this tutorial. If you have any questions or comments, please let me know in the comment section below or send me an email. 


<h3> Bibliography </h3>
<p>
*Li H., Handsaker B., Wysoker A., Fennell T., Ruan J., Homer N., Marth G., Abecasis G., Durbin R. and 1000 Genome Project Data Processing Subgroup (2009) The Sequence alignment/map (SAM) format and SAMtools. Bioinformatics, 25, 2078-9. [PMID: 19505943]

* Li H. A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics. 2011 Nov 1;27(21):2987-93. Epub 2011 Sep 8. [PMID: 21903627]

* Danecek P., Schiffels S., Durbin R. Multiallelic calling model in bcftools (-m), samtools.github.io/bcftools/call-m.pdf

* Li H. Improving SNP discovery by base alignment quality. Bioinformatics. 2011 Apr 15;27(8):1157-8. doi: 10.1093/bioinformatics/btr076. Epub 2011 Feb 13. [PMID: 21320865]

* Durbin R. Segregation based metric for variant call QC, samtools.github.io/bcftools/rd-SegBias.pdf

* Li H, Mathematical Notes on SAMtools Algorithms, www.broadinstitute.org/gatk/media/docs/Samtools.pdf
</p>

<a href="https://click.linksynergy.com/fs-bin/click?id=876kEArXFCo&offerid=659193.10001503&subid=0&type=4" rel="nofollow"><IMG border="0"   alt="Microsoft" src="https://ad.linksynergy.com/fs-bin/show?id=876kEArXFCo&bids=659193.10001503&subid=0&type=4&gridnum=0"></a>


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-123359651-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-123359651-1');
</script>

<script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script>
  (adsbygoogle = window.adsbygoogle || []).push({
    google_ad_client: "ca-pub-5126027065024936",
    enable_page_level_ads: true
  });
</script>

