---
layout: post
title: "Publication: Haplotyping the Vitis collinear core genome with rhAmpSeq improves marker transferability in a diverse genus"
tags: [rhAmpSeq,Vitis, Universal Marker, Genetic diversiy, Amplicon sequencing]
image: /image/rhAmpSeq/circos.png
share-img: /image/rhAmpSeq/circos.png
---

<h2>Abstract:</h2>
<p>Transferable DNA markers are essential for breeding and genetics. Grapevine (Vitis) breeders utilize disease resistance alleles from congeneric species ~20 million years divergent, but existing Vitis marker platforms have cross-species transfer rates as low as 2%. Here, we apply a marker strategy targeting the inferred Vitis core genome. Incorporating seven linked-read de novo assemblies and three existing assemblies, the Vitis collinear core genome is estimated to converge at 39.8â€‰Mb (8.67% of the genome). Adding shotgun genome sequences from 40 accessions enables identification of conserved core PCR primer binding sites flanking polymorphic haplotypes with high information content. From these target regions, we develop 2,000 rhAmpSeq markers as a PCR multiplex and validate the panel in four biparental populations spanning the diversity of the Vitis genus, showing transferability increases to 91.9%. This marker development strategy should be widely applicable for genetic studies in many taxa, particularly those ~20 million years divergent.
</p>
<p> Link: <a href="https://www.nature.com/articles/s41467-019-14280-1"> Click here for the online version of the paper</a></p>

<div id="seed">
<script src="/js/PDFObject.js"></script>
<script>PDFObject.embed("/image/rhAmpSeq/published_012120_NatureComm_rhAmpSeq_s41467-019-14280-1.pdf", "#seed");</script>
<style>
.pdfobject-container { height: 85rem; border: 1rem solid rgba(0,0,0,.1); }
</style>
</div>

If you have any comments or suggestions, please let comment below or send me an email. 
